### PR TITLE
build(server): Update ts compiler && moving app files to src folder

### PR DESCRIPTION
**Pull request by salsifitewn**
Move app files to ./src
Add new npm commands (ex `npm run dev` to watch for files changes)
  
References:
- https://github.com/Microsoft/TypeScript-Node-Starter#configuring-typescript-compilation
- https://www.typescriptlang.org/tsconfig

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -27,6 +30,7 @@
   "devDependencies": {
     "@types/express": "^4.17.9",
     "@types/node": "^14.14.14",
+    "concurrently": "^5.3.0",
```
Comment (ninalem) [inline server/package.json:33]: Why concurrently? Isn't it pretty much the same than nodemon?

File: server/tsconfig.json [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
-    "baseUrl": "./lib"
+    "target": "ES2020",
+    "outDir": "dist",
+    "baseUrl": ".",
+    "paths": {
+      "*": ["node_modules/*", "src/types/*"]
+    }
   },
-  "include": ["**/*.ts"],
+  "include": ["src/**/*"],
```
Comment (ninalem) [inline server/tsconfig.json:16]: Really nice to move everything to a src folder it was needed thanks

**Review by ninalem — APPROVED** [review]
I'm just not sure about the real use of concurrently, shouldn't we chose between it or nodemon rather than have multiple dependencies tending to do similar things ?
Then real nice config files and it was great to move files to a src folder. It seems all good to me!

**Comment by salsifitewn** [comment]
We need to watch two types of files ts and js. So we use two process at the same time to look for modified files. tsc for .ts and then nodemon for .js in the dist folder. Concurrently is used to launch  the two process with a pretty output and and a better error management 
https://www.npmjs.com/package/concurrently

**Comment by salsifitewn** [comment]
Closed PR due to another config implementation (#50 )

**Pull request closed without merging**